### ISSUE TITLE
typeset plugin reads headings in code blocks

### ISSUE THREAD

**Issue by pawamoy** [squidfunk/mkdocs-material#5446]
### Context

I'm writing some docs for mkdocstrings where we show how you can inject documentation for an object directly in a heading line:

```md
## ::: path.to.object
```

### Bug description

The typeset plugin naively picks up Markdown lines starting with `#` without checking if they're part of a fenced code block. It then converts those lines using the extension configured in mkdocs.yml, in this case mkdocstrings, which triggers all the code scanning + HTML rendering machinery, causing issues.

### Related links

https://github.com/squidfunk/mkdocs-material-insiders/blob/d82bb8e0a5ef39bed97e6a8849b8146dfd7afbb1/material/plugins/typeset/plugin.py#L74-L77

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/11362068/example.zip)

### Steps to reproduce

```
cd repro-typeset
python -m venv .venv
. .venv/bin/activate
python -m pip install mkdocs-material mkdocstrings[python]
mkdocs build
```

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by pawamoy** [comment]
Here's a suggestion (probably naive as well):

```python
headings = []
for line in page.markdown.split("\n"):
    if line.startswith("```"):
        if in_code_block:
            end_ticks = len(line) - len(line.lstrip("`"))
            if end_ticks == start_ticks:
                in_code_block = False
        else:
            start_ticks = len(line) - len(line.lstrip("`"))
            in_code_block = True
    elif line.startswith("#") and not in_code_block:
        headings.append(line)

html = self.markdown.convert("\n".join(headings))
```

**Comment by squidfunk** [comment]
The issue is resolved, as discussed on Gitter with @pawamoy.

**Comment by squidfunk** [comment]
Released as part of [9.1.12-insiders-4.34.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.34.0).
